### PR TITLE
Port hardshrink_backward and softshrink_backward to structured

### PR DESCRIPTION
**Pull request by Freey0**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58661 Port silu_backward to structured
* #58665 Port gelu_backward to structured
* #58664 Port gelu to structured
* #58663 Port hardshrink to structured
* **#58662 Port hardshrink_backward and softshrink_backward to structured**

Differential Revision: [D28572532](https://our.internmc.facebook.com/intern/diff/D28572532)

### CODE REVIEW

File: aten/src/ATen/native/Activation.cpp [hunk 1]
```diff
@@ -123,6 +123,13 @@ TORCH_META_FUNC(hardsigmoid_backward) (const Tensor& grad_output, const Tensor&
   build_binary_op(maybe_get_output(), grad_output, self);
 }
 
+TORCH_META_FUNC(hardshrink_backward) (
+  const Tensor & grad, const Tensor & self, const Scalar& lambd
+) {
+  set_output(0, self.sizes(), {}, self.options().memory_format(LEGACY_CONTIGUOUS_MEMORY_FORMAT), {});
```
Comment (ezyang) [inline aten/src/ATen/native/Activation.cpp:129]: Hmm, this is weird......

Comment (ezyang) [reply aten/src/ATen/native/Activation.cpp:129]: Yeah, I don't think this line is necessary

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572532).

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572532).

**Review by ezyang — APPROVED** [review]

**Comment by Freey0** [comment]
Test failure is not relevant

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572532).

**Pull request closed without merging**